backend/core/managers.py

```
@@ -0,0 +1,10 @@
+from django.db import IntegrityError
+from django.db.models import Manager as BaseModelManager
+from rest_framework.exceptions import ValidationError
+
+class ModelManager(BaseModelManager):
+    def create_with_validation(self, context_message: str = "", **kwargs):
+        try:
+            return self.create(**kwargs)
+        except (ValidationError, IntegrityError) as e:
+            raise ValidationError(f"{context_message}: {e}")
```

Ruff (0.13.1)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)